framework/finalizer.go

```diff
@@ -24,30 +24,57 @@ type patchSpec struct {
 
 func (f *Framework) addFinalizer(obj interface{}) (stopReconciliation bool, err error) {
 	restClient := f.k8sClient.CoreV1().RESTClient()
-	patch, path, result, err := createAddFinalizerPatch(obj, f.name)
-	if err != nil {
-		return false, microerror.Mask(err)
-	}
-	if patch == nil {
-		return result, err
-	}
-	p, err := json.Marshal(patch)
+	// We get the accessor of the object which we got passed from the framework.
+	accessor, err := meta.Accessor(obj)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
+	path := accessor.GetSelfLink()
+	// Continuing with reconciliation is set as a default here.
+	stopReconciliation = false
+
 	operation := func() error {
-		res := restClient.Patch(types.JSONPatchType).AbsPath(path).Body(p).Do()
-		if res.Error() != nil {
+		// We get an up to date version of our object from k8s.
+		res := restClient.Get().AbsPath(path).Do()
```

Comment:
This is likely the most important change: `RESTClient` lets us retrieve the entire object with just the `SelfLink`.

Reply:
To make less pressure on the api we can use field selector and take only finalizers

Reply:
I couldn't find a way to get the field selector here. Is there a documented way to do this?

Reply:
Ignore it for now. IMO the pressure is more on the bandwidth side because the data has to be fetched from storage anyway and I doubt they store every tiny bit separately.